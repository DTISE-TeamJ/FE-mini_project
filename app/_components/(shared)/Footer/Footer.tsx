import Link from "next/link";
import React from "react";
import { FaArrowRight, FaXTwitter } from "react-icons/fa6";
import {
  FaEnvelope,
  FaFacebookF,
  FaYoutube,
  FaInstagram,
} from "react-icons/fa";
import {
  FooterContainer,
  Row,
  Column,
  Heading,
  Underline,
  EmailId,
  Logo,
  List,
  Form,
  SocialIcons,
  Divider,
  Copyright,
} from "./style";
// import LogoImg from "@/assets/scuderia-alpha-tauri.svg";
// import LogoImg2 from "@/assets/voila2.png";
import Image from "next/image";

const Footer: React.FC = () => {
  const date = new Date();
  const year = date.getFullYear();

  return (
    <FooterContainer>
      <Row>
        <Column>
          <h1 className="text-2xl font-bold text-white mb-4">PEACHES.</h1>
          {/* <Image
            src={LogoImg2}
            alt="test-img"
            className="top-[-100px] h-[-100px] w-[-100px] relative"
          /> */}
          <p>
            Lorem ipsum dolor sit amet consectetur adipisicing elit. Ab, quos.
            Iure, velit? Harum commodi impedit itaque, iure adipisci odio
            deleniti eaque, ducimus assumenda optio ullam. Consequatur nesciunt
            aperiam dolores cum ipsum voluptas impedit.
          </p>
        </Column>
        <Column>
          <Heading>
            Office
            <Underline>
              <span></span>
            </Underline>
          </Heading>
          <p>Abbey Road</p>
          <p>London NW8 9AY</p>
          <p>United Kingdom</p>
          <EmailId>belandajuara@gmail.co.nl</EmailId>
          <h4>+44 20 1234 5678</h4>
        </Column>
        <Column>
          <Heading>
            Links
            <Underline>
              <span></span>
            </Underline>
          </Heading>
          <List>
            <li>
              <Link href="/">Home</Link>
            </li>
            <li>
              <Link href="/">Services</Link>
            </li>
            <li>
              <Link href="/">About Us</Link>
            </li>
            <li>
              <Link href="/">Features</Link>
            </li>
            <li>
              <Link href="/">Contacts</Link>
            </li>
          </List>
        </Column>
        <Column>
          <Heading>
            Newsletter
            <Underline>
              <span></span>
            </Underline>
          </Heading>
          <Form>
            <FaEnvelope className="far fa-envelope" />
            <input type="email" placeholder="Enter your email id" required />
            <Link href="/">
              <FaArrowRight className="fas fa-arrow-right" />
            </Link>
          </Form>
          <SocialIcons>
            <Link href="https://www.facebook.com/" target="_blank">
              <FaFacebookF className="fab fa-facebook" />
            </Link>
            <Link href="https://x.com/" target="_blank">
              <FaXTwitter className="fab fa-twitter" />
            </Link>
            <Link href="https://www.youtube.com/" target="_blank">
              <FaYoutube className="fab fa-youtube" />
            </Link>
            <Link href="https://www.instagram.com/" target="_blank">
              <FaInstagram className="fab fa-instagram" />
            </Link>
          </SocialIcons>
        </Column>
      </Row>
      <Divider />
      <Copyright>
        Mini Project - Team J &copy; {year} - All Rights Reserved
      </Copyright>
    </FooterContainer>
  );
};

export default Footer;
