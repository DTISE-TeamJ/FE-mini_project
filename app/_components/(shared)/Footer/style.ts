"use client";

import styled from "styled-components";

export const FooterContainer = styled.footer`
  font-family: "Poppins", sans-serif;
  width: 100%;
  bottom: 0;
  background: linear-gradient(to right, #00093c, #2d0b00);
  color: #fff;
  padding: 100px 0 30px;
  font-size: 13px;
  line-height: 20px;

  @media (max-width: 700px) {
    bottom: unset;
  }
`;

export const Row = styled.div`
  width: 85%;
  margin-left: 16px;
  margin-right: 16px;
  display: flex;
  flex-wrap: wrap;
  align-items: flex-start;
  justify-content: space-between;

  @media (min-width: 768px) {
    margin-left: auto;
    margin-right: auto;
  }
`;

export const Column = styled.div`
  flex-basis: 25%;
  padding: 10px;

  &:nth-child(2),
  &:nth-child(3) {
    flex-basis: 15%;

    @media (max-width: 700px) {
      flex-basis: 100%;
    }
  }

  @media (max-width: 700px) {
    flex-basis: 100%;
  }
`;

export const Heading = styled.h3`
  width: -moz-fit-content;
  width: fit-content;
  margin-bottom: 40px;
  position: relative;
`;

export const Underline = styled.div`
  width: 100%;
  height: 5px;
  background: #767676;
  border-radius: 3px;
  position: absolute;
  top: 25px;
  left: 0;
  overflow: hidden;

  span {
    width: 15px;
    height: 100%;
    background: #fff;
    border-radius: 3px;
    position: absolute;
    top: 0;
    left: 10px;
    animation: moving 2s linear infinite;
  }

  @keyframes moving {
    0% {
      left: -20px;
    }
    100% {
      left: 100%;
    }
  }
`;

export const EmailId = styled.p`
  width: -moz-fit-content;
  width: fit-content;
  border-bottom: 1px solid #ccc;
  margin: 20px 0;
`;

export const Logo = styled.img`
  width: 80px;
  margin-bottom: 30px;
`;

export const List = styled.ul`
  list-style: none;

  li {
    margin-bottom: 12px;

    a {
      text-decoration: none;
      color: inherit;
    }
  }
`;

export const Form = styled.form`
  padding-bottom: 15px;
  display: flex;
  justify-content: space-between;
  align-items: center;
  border-bottom: 1px solid #ccc;
  margin-bottom: 50px;

  .far {
    font-size: 18px;
    margin-right: 10px;
  }

  input {
    width: 100%;
    background: transparent;
    color: #ccc;
    border: 0;
    outline: none;
  }

  .fas {
    font-size: 16px;
    cursor: pointer;
  }
`;

export const SocialIcons = styled.div`
  display: flex;

  .fab {
    width: 30px;
    height: 30px;
    border-radius: 50%;
    padding: 8px;
    background: #fff;
    text-align: center;
    line-height: 30px;
    font-size: 10px;
    color: #000;
    margin-right: 10px;
    cursor: pointer;
  }

  @media (max-width: 576px) {
    justify-content: space-evenly;
    gap: 20px;

    .fab {
      margin-right: 0;
    }
  }
`;

export const Divider = styled.hr`
  width: 90%;
  border: 0;
  border-bottom: 1px solid #ccc;
  margin: 20px auto;
`;

export const Copyright = styled.p`
  text-align: center;
`;
