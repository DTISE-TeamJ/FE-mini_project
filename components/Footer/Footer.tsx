"use client";

import Link from "next/link";
import React from "react";
import { FaArrowRight } from "react-icons/fa6";
import {
  FaEnvelope,
  FaFacebookF,
  FaTwitter,
  FaYoutube,
  FaWhatsapp,
} from "react-icons/fa";
import {
  FooterContainer,
  Row,
  Column,
  Heading,
  Underline,
  EmailId,
  Logo,
  List,
  Form,
  SocialIcons,
  Divider,
  Copyright,
} from "./style";

const Footer: React.FC = () => {
  const date = new Date();
  const year = date.getFullYear();

  return (
    <FooterContainer>
      <Row>
        <Column>
          <Logo src="" alt="test img" />
          <p>
            Lorem ipsum dolor sit amet consectetur adipisicing elit. Ab, quos.
            Iure, velit? Harum commodi impedit itaque, iure adipisci odio
            deleniti eaque, ducimus assumenda optio ullam. Consequatur nesciunt
            aperiam dolores cum ipsum voluptas impedit.
          </p>
        </Column>
        <Column>
          <Heading>
            Office
            <Underline>
              <span></span>
            </Underline>
          </Heading>
          <p>ITPL Road</p>
          <p>Whitefield, Bangalore</p>
          <p>Karnataka, PIN 560066, India</p>
          <EmailId>belandajuara@gmail.co.nl</EmailId>
          <h4>+123456789</h4>
        </Column>
        <Column>
          <Heading>
            Links
            <Underline>
              <span></span>
            </Underline>
          </Heading>
          <List>
            <li>
              <Link href="/">Home</Link>
            </li>
            <li>
              <Link href="/">Services</Link>
            </li>
            <li>
              <Link href="/">About Us</Link>
            </li>
            <li>
              <Link href="/">Features</Link>
            </li>
            <li>
              <Link href="/">Contacts</Link>
            </li>
          </List>
        </Column>
        <Column>
          <Heading>
            Newsletter
            <Underline>
              <span></span>
            </Underline>
          </Heading>
          <Form>
            <FaEnvelope className="far fa-envelope" />
            <input type="email" placeholder="Enter your email id" required />
            <Link href="/">
              <FaArrowRight className="fas fa-arrow-right" />
            </Link>
          </Form>
          <SocialIcons>
            <FaFacebookF className="fab fa-facebook" />
            <FaTwitter className="fab fa-twitter" />
            <FaYoutube className="fab fa-youtube" />
            <FaWhatsapp className="fab fa-whatsapp" />
          </SocialIcons>
        </Column>
      </Row>
      <Divider />
      <Copyright>
        Mini Project - Team J &copy; {year} - All Rights Reserved
      </Copyright>
    </FooterContainer>
  );
};

export default Footer;
