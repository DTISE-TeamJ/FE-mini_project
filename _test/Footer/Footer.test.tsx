import React from "react";
import { render, screen } from "@testing-library/react";
// import Footer from "../Footer";
import Footer from "app/_components/(shared)/Footer/Footer";
import "@testing-library/jest-dom";

jest.mock("next/link", () => {
  return ({ children }: { children: React.ReactNode }) => {
    return children;
  };
});

describe("Footer Component", () => {
  it("renders correctly", () => {
    const { container } = render(<Footer />);
    expect(container).toMatchSnapshot();
  });

  it("renders the company name", () => {
    render(<Footer />);
    const heading = screen.getByTestId("headingFooter");
    expect(heading).toBeInTheDocument();
    expect(heading).toHaveTextContent("PEACHES.");
  });

  it("displays the correct office address", () => {
    render(<Footer />);
    expect(screen.getByText("Abbey Road")).toBeInTheDocument();
    expect(screen.getByText("London NW8 9AY")).toBeInTheDocument();
    expect(screen.getByText("United Kingdom")).toBeInTheDocument();
  });

  it("shows the correct email and phone number", () => {
    render(<Footer />);
    expect(screen.getByText("belandajuara@gmail.co.nl")).toBeInTheDocument();
    expect(screen.getByText("+44 20 1234 5678")).toBeInTheDocument();
  });

  it("renders all navigation links", () => {
    render(<Footer />);
    const links = ["Home", "Services", "About Us", "Features", "Contacts"];
    links.forEach((link) => {
      expect(screen.getByText(link)).toBeInTheDocument();
    });
  });

  it("includes a newsletter form", () => {
    render(<Footer />);
    expect(
      screen.getByPlaceholderText("Enter your email id")
    ).toBeInTheDocument();
  });

  // it("displays social media icons", () => {
  //   render(<Footer />);
  //   expect(screen.getByRole("link", { name: /facebook/i })).toBeInTheDocument();
  //   expect(screen.getByRole("link", { name: /twitter/i })).toBeInTheDocument();
  //   expect(screen.getByRole("link", { name: /youtube/i })).toBeInTheDocument();
  //   expect(
  //     screen.getByRole("link", { name: /instagram/i })
  //   ).toBeInTheDocument();
  // });

  it("shows the correct copyright text", () => {
    render(<Footer />);
    const currentYear = new Date().getFullYear();
    expect(
      screen.getByText(
        `Mini Project - Team J Â© ${currentYear} - All Rights Reserved`
      )
    ).toBeInTheDocument();
  });
});
