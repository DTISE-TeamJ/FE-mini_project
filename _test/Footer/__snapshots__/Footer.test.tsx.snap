// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer Component renders correctly 1`] = `
<div>
  <footer
    class="sc-a0c0ccd3-0 boMdIj"
  >
    <div
      class="sc-a0c0ccd3-1 iEzgck"
    >
      <div
        class="sc-a0c0ccd3-2 hhEJjG"
      >
        <h1
          class="text-2xl font-bold text-white mb-4"
          data-testid="headingFooter"
        >
          PEACHES.
        </h1>
        <p>
          Lorem ipsum dolor sit amet consectetur adipisicing elit. Ab, quos. Iure, velit? Harum commodi impedit itaque, iure adipisci odio deleniti eaque, ducimus assumenda optio ullam. Consequatur nesciunt aperiam dolores cum ipsum voluptas impedit.
        </p>
      </div>
      <div
        class="sc-a0c0ccd3-2 hhEJjG"
      >
        <h3
          class="sc-a0c0ccd3-3 jbuLyH"
        >
          Office
          <div
            class="sc-a0c0ccd3-4 gDvAqg"
          >
            <span />
          </div>
        </h3>
        <p>
          Abbey Road
        </p>
        <p>
          London NW8 9AY
        </p>
        <p>
          United Kingdom
        </p>
        <p
          class="sc-a0c0ccd3-5 ZgwFy"
        >
          belandajuara@gmail.co.nl
        </p>
        <h4>
          +44 20 1234 5678
        </h4>
      </div>
      <div
        class="sc-a0c0ccd3-2 hhEJjG"
      >
        <h3
          class="sc-a0c0ccd3-3 jbuLyH"
        >
          Links
          <div
            class="sc-a0c0ccd3-4 gDvAqg"
          >
            <span />
          </div>
        </h3>
        <ul
          class="sc-a0c0ccd3-7 kAaXYz"
        >
          <li>
            Home
          </li>
          <li>
            Services
          </li>
          <li>
            About Us
          </li>
          <li>
            Features
          </li>
          <li>
            Contacts
          </li>
        </ul>
      </div>
      <div
        class="sc-a0c0ccd3-2 hhEJjG"
      >
        <h3
          class="sc-a0c0ccd3-3 jbuLyH"
        >
          Newsletter
          <div
            class="sc-a0c0ccd3-4 gDvAqg"
          >
            <span />
          </div>
        </h3>
        <form
          class="sc-a0c0ccd3-8 dquFJk"
        >
          <svg
            class="far fa-envelope"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 512 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M502.3 190.8c3.9-3.1 9.7-.2 9.7 4.7V400c0 26.5-21.5 48-48 48H48c-26.5 0-48-21.5-48-48V195.6c0-5 5.7-7.8 9.7-4.7 22.4 17.4 52.1 39.5 154.1 113.6 21.1 15.4 56.7 47.8 92.2 47.6 35.7.3 72-32.8 92.3-47.6 102-74.1 131.6-96.3 154-113.7zM256 320c23.2.4 56.6-29.2 73.4-41.4 132.7-96.3 142.8-104.7 173.4-128.7 5.8-4.5 9.2-11.5 9.2-18.9v-19c0-26.5-21.5-48-48-48H48C21.5 64 0 85.5 0 112v19c0 7.4 3.4 14.3 9.2 18.9 30.6 23.9 40.7 32.4 173.4 128.7 16.8 12.2 50.2 41.8 73.4 41.4z"
            />
          </svg>
          <input
            placeholder="Enter your email id"
            required=""
            type="email"
          />
          <svg
            class="fas fa-arrow-right"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 448 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M438.6 278.6c12.5-12.5 12.5-32.8 0-45.3l-160-160c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3L338.8 224 32 224c-17.7 0-32 14.3-32 32s14.3 32 32 32l306.7 0L233.4 393.4c-12.5 12.5-12.5 32.8 0 45.3s32.8 12.5 45.3 0l160-160z"
            />
          </svg>
        </form>
        <div
          class="sc-a0c0ccd3-9 eUMroD"
        >
          <svg
            class="fab fa-facebook"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 320 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M279.14 288l14.22-92.66h-88.91v-60.13c0-25.35 12.42-50.06 52.24-50.06h40.42V6.26S260.43 0 225.36 0c-73.22 0-121.08 44.38-121.08 124.72v70.62H22.89V288h81.39v224h100.17V288z"
            />
          </svg>
          <svg
            class="fab fa-twitter"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 512 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M389.2 48h70.6L305.6 224.2 487 464H345L233.7 318.6 106.5 464H35.8L200.7 275.5 26.8 48H172.4L272.9 180.9 389.2 48zM364.4 421.8h39.1L151.1 88h-42L364.4 421.8z"
            />
          </svg>
          <svg
            class="fab fa-youtube"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 576 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M549.655 124.083c-6.281-23.65-24.787-42.276-48.284-48.597C458.781 64 288 64 288 64S117.22 64 74.629 75.486c-23.497 6.322-42.003 24.947-48.284 48.597-11.412 42.867-11.412 132.305-11.412 132.305s0 89.438 11.412 132.305c6.281 23.65 24.787 41.5 48.284 47.821C117.22 448 288 448 288 448s170.78 0 213.371-11.486c23.497-6.321 42.003-24.171 48.284-47.821 11.412-42.867 11.412-132.305 11.412-132.305s0-89.438-11.412-132.305zm-317.51 213.508V175.185l142.739 81.205-142.739 81.201z"
            />
          </svg>
          <svg
            class="fab fa-instagram"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 448 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M224.1 141c-63.6 0-114.9 51.3-114.9 114.9s51.3 114.9 114.9 114.9S339 319.5 339 255.9 287.7 141 224.1 141zm0 189.6c-41.1 0-74.7-33.5-74.7-74.7s33.5-74.7 74.7-74.7 74.7 33.5 74.7 74.7-33.6 74.7-74.7 74.7zm146.4-194.3c0 14.9-12 26.8-26.8 26.8-14.9 0-26.8-12-26.8-26.8s12-26.8 26.8-26.8 26.8 12 26.8 26.8zm76.1 27.2c-1.7-35.9-9.9-67.7-36.2-93.9-26.2-26.2-58-34.4-93.9-36.2-37-2.1-147.9-2.1-184.9 0-35.8 1.7-67.6 9.9-93.9 36.1s-34.4 58-36.2 93.9c-2.1 37-2.1 147.9 0 184.9 1.7 35.9 9.9 67.7 36.2 93.9s58 34.4 93.9 36.2c37 2.1 147.9 2.1 184.9 0 35.9-1.7 67.7-9.9 93.9-36.2 26.2-26.2 34.4-58 36.2-93.9 2.1-37 2.1-147.8 0-184.8zM398.8 388c-7.8 19.6-22.9 34.7-42.6 42.6-29.5 11.7-99.5 9-132.1 9s-102.7 2.6-132.1-9c-19.6-7.8-34.7-22.9-42.6-42.6-11.7-29.5-9-99.5-9-132.1s-2.6-102.7 9-132.1c7.8-19.6 22.9-34.7 42.6-42.6 29.5-11.7 99.5-9 132.1-9s102.7-2.6 132.1 9c19.6 7.8 34.7 22.9 42.6 42.6 11.7 29.5 9 99.5 9 132.1s2.7 102.7-9 132.1z"
            />
          </svg>
        </div>
      </div>
    </div>
    <hr
      class="sc-a0c0ccd3-10 bmYeJf"
    />
    <p
      class="sc-a0c0ccd3-11 cCVTKq"
    >
      Mini Project - Team J Â© 
      2024
       - All Rights Reserved
    </p>
  </footer>
</div>
`;
